### PR TITLE
Replace JAXopt with Optax

### PR DESCRIPTION
**Pull request by tzunghanjuang**
**Context:** 

Google is merging JAXopt into Optax. However, Some JAXopt classes and methods (e.g., ``jaxopt.GradientDescent``) still have no their counterparts in Optax.  Moreover, the interface of Optax's methods is different from that of JAXopt's.

**Description of the Change:**

- Replace ``jaxopt.GradientDescent`` with ``optax.sgd``.
- Remove the examples with ``jaxopt.GradientDescent.run`` method since it has no counterpart in optax.

**Benefits:**

**Possible Drawbacks:**

- The results of ``optax.sgd`` can be different from those of ``jaxopt.GradientDescent``.
- ~~I cannot reproduce some results (which are not related to jaxopt and optax) in demos. Might be the issues of environment setup and random seeds.~~

**Related GitHub Issues:** https://github.com/PennyLaneAI/catalyst/issues/499
[sc-65610]

### CODE REVIEW

**Comment by dime10** [comment]
@josh146 @isaacdevlugt

File: demos/adaptive_circuits_demo.ipynb [hunk 1]
```diff
@@ -978,7 +986,7 @@
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "params_doubles: [ 0.04758607 -0.09855445  0.09865958  0.05330061  0.05331783  0.22938123]\n"
```
Comment (josh146) [inline demos/adaptive_circuits_demo.ipynb:981]: Hey @tzunghanjuang, it surprises me that the new output here shows all the same value in the list 🤔 Did something change/go wrong here?

File: doc/dev/quick_start.rst [hunk 2]
```diff
@@ -741,12 +744,3 @@ as well as vectorized using ``jax.vmap``:
...
 Array([1.32269195, 1.53905377], dtype=float64)
-
-In particular, this allows for a reduction in boilerplate when using
-JAX-compatible optimizers such as ``jaxopt``:
-
->>> opt = jaxopt.GradientDescent(cost_fn)
->>> params = jnp.array([0.1, 0.2, 0.3])
->>> (final_params, _) = jax.jit(opt.run)(params)
->>> final_params
-Array([-0.00320799,  0.03475223,  0.29362844], dtype=float64)
```
Comment (josh146) [inline doc/dev/quick_start.rst:752]: @tzunghanjuang is there an optax version we could replace this with?

File: demos/adaptive_circuits_demo.ipynb (hunk 1 above)
Comment (isaacdevlugt) [reply demos/adaptive_circuits_demo.ipynb:981]: @tzunghanjuang did mention this in possible drawbacks

> I cannot reproduce some results (which are not related to jaxopt and optax) in demos. Might be the issues of environment setup and random seeds.

Maybe this is an example?

**Comment by isaacdevlugt** [comment]
Thanks @tzunghanjuang! Where are we getting the info that `JAXOpt` is merging into `Optax`?

File: doc/dev/quick_start.rst (hunk 2 above)
Comment (tzunghanjuang) [reply doc/dev/quick_start.rst]: I have wrapped up a function for that in this test.

File: doc/dev/quick_start.rst [hunk 3]
```diff
@@ -742,11 +745,30 @@ as well as vectorized using ``jax.vmap``:
 >>> jax.vmap(cost_fn)(jnp.array([[0.1, 0.2, 0.3], [0.4, 0.5, 0.6]]))
 Array([1.32269195, 1.53905377], dtype=float64)
 
+
 In particular, this allows for a reduction in boilerplate when using
-JAX-compatible optimizers such as ``jaxopt``:
+JAX-compatible optimizers such as ``optax``.
```
Comment (josh146) [inline doc/dev/quick_start.rst:750]: @tzunghanjuang I now see why you deleted this; it's actually no less boilerplate 😆 

Feel free to delete this section if you think it makes sense.

**Review by josh146 — APPROVED** [review]
Nice work @tzunghanjuang!

File: doc/dev/quick_start.rst (hunk 3 above)
Comment (tzunghanjuang) [reply doc/dev/quick_start.rst]: Thank you. I have removed this part.

**Pull request merged**